kdb/key.go

```diff
@@ -34,16 +35,21 @@ type Key interface {
 
 	Value() string
 	Boolean() bool
+	// TODO REVIEW: other types (or move all types to higher-level)?
```

Comment:
I've added  `keyGetString` -> `Value()` and `keyGetBinary` -> Bytes()` since they exist in `kdb.h`. Maybe I should remove `Boolean()` to prevent straying to far from the kdb.h?

Reply:
It is good to have Boolean and the other types. But maybe they should be in the higher-level API.

Reply:
Ok - I will remove the boolean type

Reply:
You can simply move it to the high-level API.

Reply:
The high-level api is virtually none-existant at the moment and I'm not sure if there's enough time to create the binding for it.

Reply:
As discussed, the high-level go API does not necessarily need to be based on the high-level C API.

Having get/set for all types of go is a first step for a high-level go API.